ec/ecresource/trafficfilterassocresource/resource_test.go

```diff
@@ -0,0 +1,324 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package trafficfilterassocresource_test
+
+import (
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/api/mock"
+	"github.com/elastic/terraform-provider-ec/ec"
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"net/url"
+	"regexp"
+	"testing"
+
+	r "github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestResourceTrafficFilterAssoc(t *testing.T) {
+
+	r.UnitTest(t, r.TestCase{
+		ProtoV5ProviderFactories: protoV5ProviderFactoriesWithMockClient(
+			api.NewMock(
+				createResponse(),
+				readResponse(),
+				readResponse(),
+				readResponse(),
+				readResponse(),
+				deleteResponse(),
+			),
+		),
+		Steps: []r.TestStep{
+			{ // Create resource
+				Config: trafficFilterAssoc,
+				Check:  checkResource(),
+			},
+			{ // Ensure that it can be successfully read
+				PlanOnly: true,
+				Config:   trafficFilterAssoc,
+				Check:    checkResource(),
+			},
+			{ // Delete resource
```

Comment:
Why the step is needed?

Reply:
Or can we check it somehow?

Reply:
I wanted to make testing destroy explicit. I will add a check for it.

Reply:
Like I wrote - the check is performed via the RequestAssertion.